tests/cases/compiler/exportAsNamespace_augment.ts

```diff
@@ -0,0 +1,22 @@
+// @Filename: /a.d.ts
+export as namespace a;
+export const x = 0;
+export const conflict = 0;
+
+// @Filename: /b.ts
+import * as a2 from "./a";
+
+declare global {
+    namespace a {
```

Comment:
I'm not convinced we need this ability. Does this PR need to exist for the `declare module "./a"` version to work?

Reply:
Currently, if you have:
**a.d.ts**
```ts
export const x = 0;
export as namespace a;
```

**b.ts**
```ts
export {};
declare module "./a" {
    export const y = 0;
}
```
**c.ts**
```ts
a.y;
```

It will fail in `c.ts`. But there is no failure for `import { y } from "./a";` -- the module got augmented but the namespace didn't.